### ISSUE TITLE
Crash on Android because ScreenTransition do not inherit from Serializable

### ISSUE THREAD

**Issue by beyondeye** [adrielcafe/voyager#438]
When using the new Per Screen transitions API and implementing ``enter()`` and ``exit()`` overrides with a delegate like described 
in the documentation
```kotlin
class SlideScreen : Screen, ScreenTransition by SlideTransition() {
//...
}
```
The application will crash on Android (and probably also on other platforms) if ``class SlideTransition`` is not defined as a ``Serializable`` class

**Comment by DevSrSouza** [comment]
SlideTransition not being at the current library, I assume this is a docs update requirement, right? Just want to understand if is a docs problem or library problem.

**Comment by beyondeye** [comment]
Yes It should be documented that transition classes that are used to delegate implementation of Screen enter() and exit() methods must be defined as Serializable, the same as Screen classes

**Comment by shpasha** [comment]
> Yes It should be documented that transition classes that are used to delegate implementation of Screen enter() and exit() methods must be defined as Serializable, the same as Screen classes

I dont understand why. I use code like in docs in my project and it works. Also the code from the samples works for me on Android. 

Can you please give a more detailed example of a crash?

**Comment by shpasha** [comment]
Have you tried to implement ScreenTransition like in the sample? This code works fine for me on Android

https://github.com/adrielcafe/voyager/tree/main/samples/android/src/main/java/cafe/adriel/voyager/sample/screenTransition

**Comment by beyondeye** [comment]
The problem is on screen rotation, when the Screen class must be serialized and deserialized
